types.go

```diff
@@ -258,3 +258,76 @@ func getTimeTicks[T any](val T) (int64, error) {
 	base := time.Date(1970, time.January, 1, ti.Hour(), ti.Minute(), ti.Second(), ti.Nanosecond(), time.UTC)
 	return base.UnixMicro(), err
 }
+
+func getMappedArrayValue[T any](lt mapping.LogicalType, val T) (*mapping.Value, error) {
```

Comment:
Let's unify this function with `getMappedListValue` and maybe call it `getMappedSliceValue`. They only differ in a single line, which we could solve like in the snippet below. We could also pass `t` (`Type`) as a parameter.

```Go
	var v mapping.Value
	t := Type(mapping.GetTypeId(lt))
	if t == TYPE_ARRAY {
		v = mapping.CreateArrayValue(childType, childValues)
	} else {
		v = mapping.CreateListValue(childType, childValues)
	}
	
	return &v, nil
```